sympy/concrete/guess.py

```diff
@@ -377,3 +378,60 @@ def guess_generating_function(v, X=Symbol('x'), types=['all'], maxsqrtn=2):
                 break
 
     return result
+
+
+@public
+def guess(l, stop=False, evaluate=True, niter=None):
+    """
+    This function is adapted from the Rate.m package for Mathematica
+    written by Christian Krattenthaler.
+    It tries to guess a formula from a given sequence of rational numbers.
+
+    In order to speed up the process, the 'stop' variable allows to stop
```

Comment:
Are there examples in which the number of results will depend on the value of `stop`?